osu.Game/Screens/Select/Carousel/CarouselItem.cs

```diff
@@ -51,7 +49,7 @@ public virtual void Filter(FilterCriteria criteria)
 
         public virtual int CompareTo(FilterCriteria criteria, CarouselItem other) => ItemID.CompareTo(other.ItemID);
 
-        public int CompareTo(CarouselItem other) => CarouselYPosition.CompareTo(other.CarouselYPosition);
+        public int CompareTo(CarouselItem? other) => CarouselYPosition.CompareTo(other!.CarouselYPosition);
```

Comment:
Kind of suspect too. Should be handling null correctly if that's what `iComparable` requires.

Reply:
See

https://github.com/ppy/osu/blob/96a100d428c1a960c48369816a05696aed95aa50/osu.Game/Rulesets/RulesetInfo.cs#L46-L52

for a typical example of how this is done.